components/Tabs/tests/unit/MDCTabBarLayoutTests.m

```diff
@@ -0,0 +1,136 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <MDFInternationalization/MDFInternationalization.h>
+#import <XCTest/XCTest.h>
+
+#import "MDCItemBar.h"
+#import "MDCTabBar.h"
+
+// Returns the underlying collection view from a given tabBar. If one cannot be extracted, returns
+// nil.
+static UICollectionView *ExtractCollectionViewFromTabBar(MDCTabBar *tabBar) {
+  MDCItemBar *itemBar = nil;
+  for (UIView *subview in tabBar.subviews) {
+    if ([subview isKindOfClass:[MDCItemBar class]]) {
+      itemBar = (MDCItemBar *)subview;
+    }
+  }
+  UICollectionView *collectionView = nil;
+  for (UIView *subview in itemBar.subviews) {
+    if ([subview isKindOfClass:[UICollectionView class]]) {
+      collectionView = (UICollectionView *)subview;
+    }
+  }
+  return collectionView;
+}
+
+// Returns the visible items of the given collectionView in sorted order.
+static NSArray<UICollectionViewCell *> *SortedCellsFromCollectionView(
+    UICollectionView *collectionView) {
+  NSArray<NSIndexPath *> *indexPathsForVisibleItems = collectionView.indexPathsForVisibleItems;
+  NSArray<NSIndexPath *> *sortedIndexPaths =
+      [indexPathsForVisibleItems sortedArrayUsingSelector:@selector(compare:)];
+  NSMutableArray *sortedCells = [NSMutableArray array];
+  for (NSIndexPath *indexPath in sortedIndexPaths) {
+    UICollectionViewCell *cell = [collectionView cellForItemAtIndexPath:indexPath];
+    if (cell) {
+      [sortedCells addObject:cell];
+    }
+  }
+  return sortedCells;
+}
+
+@interface MDCTabBarLayoutTests : XCTestCase
+
+@end
+
+@implementation MDCTabBarLayoutTests {
+  MDCTabBar *_tabBar;
+}
+
+- (void)setUp {
+  [super setUp];
+
+  // Create a tabbar with some dummy items and a dummy frame.
+  _tabBar = [[MDCTabBar alloc] initWithFrame:CGRectZero];
+  CGFloat tabBarHeight = [MDCTabBar defaultHeightForItemAppearance:_tabBar.itemAppearance];
+  _tabBar.frame = CGRectMake(0, 0, 200, tabBarHeight);
+  UITabBarItem *item1 = [[UITabBarItem alloc] initWithTitle:@"first" image:nil tag:0];
+  UITabBarItem *item2 = [[UITabBarItem alloc] initWithTitle:@"second" image:nil tag:0];
+  _tabBar.items = @[ item1, item2 ];
+  [_tabBar setNeedsLayout];
+  [_tabBar layoutIfNeeded];
+}
+
+- (void)tearDown {
+  // Explictly nil out ivars so that they are properly deallocated between tests. Otherwise XCTest
+  // will keep them alive until all test cases are complete.
+  _tabBar = nil;
+  [super tearDown];
+}
+
+// Tests the default layout behavior of of the tab bar's collection view.
+- (void)testLeftToRightLayout {
+  // Given
+  UICollectionView *collectionView = ExtractCollectionViewFromTabBar(_tabBar);
+  NSArray<UICollectionViewCell *> *sortedVisibleItems =
+      SortedCellsFromCollectionView(collectionView);
+  XCTAssertEqual(sortedVisibleItems.count, 2ul);
```

Comment:
Tip: This has me worried that the helper function needs its own tests.  Please consider exiting the test early if the number of items does not equal 2, since the rest of the test is arguably invalid.

I've also filed #6037 to refactor MDCTabBar internally to allow easier testing.

Reply:
Added an early return for now and tested that it works as expected.